octodns/provider/route53.py

```diff
@@ -211,6 +468,35 @@ def __repr__(self):
                                                           self.values)
 
 
+_mod_keyer_action_order = {
+    'DELETE': '0',  # Delete things first
+    'CREATE': '1',  # Then Create things
+    'UPSERT': '2',  # Upsert things last
+}
+
+
+def _mod_keyer(mod):
+    rrset = mod['ResourceRecordSet']
+    action_order = _mod_keyer_action_order[mod['Action']]
+
+    # We're sorting by 3 "columns", the action, the rrset type, and finally the
+    # name/id of the rrset
```

Comment:
This sorting ensures that Route53 doesn't have problems with the order of the modifications we send it. We need to make sure that anything that's referenced by an Alias comes before that reference. We're doing that by sorting of types rather than explicitly looking through the aliases and targets.